core/src/main/java/org/pentaho/di/core/database/Database.java

```diff
@@ -2445,8 +2456,13 @@ public RowMetaInterface getQueryFieldsFromPreparedStatement( String sql ) throws
   }
 
   public RowMetaInterface getQueryFieldsFromDatabaseMetaData() throws Exception {
+    return this.getQueryFieldsFromDatabaseMetaData( null );
+  }
+
+  public RowMetaInterface getQueryFieldsFromDatabaseMetaData( String sql ) throws Exception {
```

Comment:
This method can't be private? Wouldn't it be better to have it as private, since that the exposed sql parameter ends up having an effect on lines 2464-2465 that may be "difficult to explain" to who ever uses this class.